terraform/main.tf

```diff
@@ -0,0 +1,73 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source = "hashicorp/azurerm"
+      version = ">= 2.49"
+    }
+  }
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "terraform_remote_state" "state" {
+  backend = "azurerm"
+  config = {
+    resource_group_name  = var.resource_group_name
+    storage_account_name = var.storage_account_name
+    container_name       = var.container_name
+    key                  = var.key
+  }
+}
```

Comment:
Is this used? It looks like you're passing in a backend access key, so there's no need for this. You're effectively doing the `When authenticating using the Access Key associated with the Storage Account:` option from [here](https://www.terraform.io/docs/language/settings/backends/azurerm.html), but configured via CLI rather than in the tf file.